tensorflow/python/training/saver.py

```diff
@@ -1070,6 +1070,10 @@ def save(self, sess, save_path, global_step=None, latest_filename=None,
     if not isinstance(sess, session.SessionInterface):
       raise TypeError("'sess' must be a Session; %s" % sess)
 
+    # Note a few lines above save_path was set to os.path.dirname(save_path)
```

Comment:
I'm a bit leery of this type of comment that references the structure of the code, since code changes over time and the comment may become outdated and confusing.

Why not move this check a few lines up, so it immediatley follows setting save_path = os.path.dirname(save_path)? The test that checks if the session is a Session can occur after the save path is setup. And the comment will no longer be necessary, since the immediately pre-ceeding line will be the save_path setup.